modules/component/src/core/cd-aware/strategy.ts

```diff
@@ -0,0 +1,207 @@
+// @Notice this part of the code is in the coalescing PR https://github.com/ngrx/platform/pull/2456
+// import { generateFrames } from '../projections/generateFrames';
+// import { coalesce } from '../operators/coalesce';
+import { defer, from, MonoTypeOperatorFunction, Observable } from 'rxjs';
+import {
+  ChangeDetectorRef,
+  ɵdetectChanges as detectChanges,
+  ɵmarkDirty as markDirty,
+} from '@angular/core';
+import { apiZonePatched, getGlobalThis, isViewEngineIvy } from '../utils';
+import { mapTo } from 'rxjs/operators';
+import { getZoneUnPatchedDurationSelector } from './duration-selector';
+
+export interface StrategyFactoryConfig {
+  cdRef: ChangeDetectorRef;
+}
+
+export interface CdStrategy<T> {
+  behaviour: () => MonoTypeOperatorFunction<T>;
+  render: () => void;
+  name: string;
+}
+
+export const DEFAULT_STRATEGY_NAME = 'native';
+export const IS_VIEW_ENGINE_IVY = isViewEngineIvy();
+
+export interface StrategySelection<U> {
+  [DEFAULT_STRATEGY_NAME]: CdStrategy<U>;
+  [key: string]: CdStrategy<U>;
```

Comment:
If we only have default, noop, global, and local, why don't we explicitly set these properties instead of using `[key: string]`?

Reply:
I thought of custom strategies. But I will set it directly for now!

Thanks!

Reply:
Fixed by implementing the suggestion of Alex